ROADMAP.md

```diff
@@ -0,0 +1,579 @@
+measure the impact of our changes on the large scale; to combat the
+perception in some circles that libp2p is "immature".
+
+### 🌐 Browser use cases
+
+**What?** This spans a number of concerns. First, we should improve
+developer experience and make libp2p more accessible in browser
+environments. This entails reducing footprint (e.g. bundle size) and
+creating examples and documentation on how to bundle js-libp2p with
+popular frontend tooling (webpack, parcel, etc.). Developers should be
+able to cherry-pick the features to incorporate in their bundle, and
+tree-shaking should produce good results.
+
+In terms of transports, hardening our WebRTC support and enabling it by
+default will bring a huge boost to general browser connectivity.
+
+Targeting the [WASM runtime](#libp2p-as-a-wasm-library)
+deserves a special mention, as we're likely to see more user interest in
+this area. Finally, we should continue supporting the
+[TCP](https://github.com/mozilla/libdweb/issues/5) and
+[UDP](https://github.com/mozilla/libdweb/issues/4) socket
+advances made by Mozilla's
+[libdweb](https://github.com/mozilla/libdweb/) team.
+
+**Why?** Browsers are and will continue to be the most used computer interface.
+Not fully supporting the browser platform will hurt most of the projects betting
+on libp2p.
+
+### 🌡️ Opt-in telemetry protocol
+
+**What?** By developing a *decentralized* telemetry protocol that nodes can opt
+into, we could build a world-wide real-time visualisation of our global libp2p
+network, à la [Torflow](https://torflow.uncharted.software), [Kaspersky's
+Cybermap](https://cybermap.kaspersky.com/), or [Checkpoint's
+Threatmap](https://threatmap.checkpoint.com/).
+
+This might surface interesting facts about the backbone of the Internet;
+perhaps even correlations with the [Submarine Cable
+Map](https://www.submarinecablemap.com/). Imagine [RIPE
+Atlas](https://atlas.ripe.net/results/maps/network-coverage/)
+but for libp2p networks.
+
+Nodes could probe and collect stats about the connections it maintains
+with other peers: latency, traceroute, IP address, etc., publishing raw
+data on a global pubsub channel like a heartbeat, where a webapp is
+listening over a websockets channel.
+
+**Why?** To gain real-time insight into the global libp2p deployment.
+
+### 📩 Message-oriented transports
+
+**What?** The data layer of the libp2p API is geared towards streams,
+and does not support message orientation. The notion of messages [has
+been discussed](https://github.com/libp2p/specs/issues/71)
+in the past, and there is even a [draft proposal for a multigram
+protocol](https://github.com/multiformats/multigram/tree/draft1).
+
+Progressing with this work entails designing the abstractions, and
+implementing, at least, a UDP transport, [as the community has
+demanded](https://github.com/libp2p/go-libp2p/issues/353).
+
+**Why?** Current the libp2p API precludes us from modelling
+message-oriented transports like UDP or Bluetooth.
```

Comment:
Newbie here but I'm not following what kind of end uses cases this would unlock.

Reply:
Enabling Bluetooth connectivity would be nice. Imagine your node discovering and connecting to all devices in your proximity, even if those devices didn't join the same network (in that case, they'd discover each other via mDNS). That said, I'm not sure if message orientation is a prerequisite for that, or if there's a way to establish a Bluetooth 
"connection". I'm a total Bluetooth newbie, so don't trust me on this ;)

In general, the appeal of message-oriented protocols is that they enable "fire and forget". I could send DHT insertion queries to peers without establishing connections to them.

Reply:
Resolving here. Feel free to un-resolve in case you want to continue the conversation.